### ISSUE TITLE
differing registry resolution does not cause cache mismatch

### ISSUE THREAD

**Issue by bmeck** [yarnpkg/yarn#1198]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

Dependency with matching version and name does not check resolved URL to ensure they are the same.

**If the current behavior is a bug, please provide the steps to reproduce.**

Install a package such as `slash@1.0.0` from the public registry in one app, then from a private registry in a different app. The resolved URL in yarn.lock does not match the cached yarn resolved URL.

```
$ cat yarn.lock | grep -A 3 slash
slash:
  version "1.0.0"
  resolved "https://registry.yarnpkg.com/slash/-/slash-1.0.0.tgz#c41f2f6c39fc16d1cd17ad4b5d896114ae470d55"

$ cat ~/.yarn-cache/npm-slash-1.0.0/.yarn-metadata.json  | grep resolved
    "resolved": "https://artifactory/artifactory/api/npm/node-virt/slash/-/slash-1.0.0.tgz#c41f2f6c39fc16d1cd17ad4b5d896114ae470d55",
```

**What is the expected behavior?**

Either: yarn-cache is updated to new resolved URL, an error is thrown, or a prompt is issued.

**Please mention your node.js, yarn and operating system version.**

```
node -v: v6.5.0
OS: OSX 15.6.0
yarn --version: 0.16.0
```

**Comment by aij** [comment]
Why would you expect anything different?

In the above example, both are the same. (c41f2f6c39fc16d1cd17ad4b5d896114ae470d55)

Or are you saying that hash is incorrect for one of the two? (Eg: If both repos are using the same package name and version number but for actually different packages.)

**Comment by bmeck** [comment]
yes, same name and version, but one has been compiled in my case so contents are different.
